include/rocksdb/utilities/stackable_db.h

```diff
@@ -404,8 +404,14 @@ class StackableDB : public DB {
 
   using DB::StartBlockCacheTrace;
   Status StartBlockCacheTrace(
-      const TraceOptions& options,
+      const TraceOptions& trace_options,
       std::unique_ptr<TraceWriter>&& trace_writer) override {
+    return db_->StartBlockCacheTrace(trace_options, std::move(trace_writer));
+  }
+
+  Status StartBlockCacheTrace(
+      const BlockCacheTraceOptions& options,
+      std::unique_ptr<BlockCacheTraceWriter>&& trace_writer) override {
```

Comment:
What if `StartBlockCacheTrace()` accepts a `TraceRecord::Handler` for users who want to do custom record tracing?

Reply:
I am not sure about the that approach as I am not familiar with the query tracing. Let me take a look at it. cc @anand1976 Do you have any thoughts on it.

Reply:
I checked the code and it seems its more involved with Handler as the process for query tracing is quite different from block cache tracing.

Reply:
> What if `StartBlockCacheTrace()` accepts a `TraceRecord::Handler` for users who want to do custom record tracing?

It's probably more general than needed for this PR's stated purpose. The downside to generalizing the write side so much is the read side can't make any assumptions, e.g., that a user can provide a `TraceReader` to read a trace. This PR doesn't handle the read side either but I think the design should allow it. However, you would need to introduce yet another independent interface (`BlockCacheTraceReader`).

What if instead we made a `TraceCodec` interface that defines the `string`<->`TraceRecord` conversion? It could be provided together with `TraceWriter` to all the write functions, and provided together with `TraceReader` to all the read functions. That way there is just one new interface for the combinations of (block cache, KV, I/O) x (read, write).

Reply:
What's the `string<->TraceRecord` conversion? How it would be used?

Reply:
For example, a `TraceCodec` would include the following functions:

```
class TraceCodec {
...
  virtual std::string Encode(const BlockCacheTraceRecord&) = 0;
  virtual BlockCacheTraceRecord Decode(const std::string&) = 0;
...
};
```

Then `BlockCacheTraceWriter` can be internal again, but needs to be configured with a `TraceCodec` that comes from `StartBlockCacheTrace()`. Say we store it in an instance variable, `trace_codec_`. Then `BlockCacheTraceWriter` can use it like this:

```
Status BlockCacheTraceWriter::WriteBlockAccess(
    const BlockCacheTraceRecord& record) {
  return trace_writer_->Write(trace_codec_->Encode(record)); 
}

Reply:
Ah ok. That's something that can be done. I will give it a shot and see how it goes.

Reply:
Just to confirm, is your client use case OK with using `TraceWriter` for writing the physical trace records? If not, something like the `BlockCacheTraceWriter` seems necessary to customize writing logical trace records.